pkg/api/routes.go

```diff
@@ -398,6 +405,116 @@ func (rh *RouteHandler) GetManifest(response http.ResponseWriter, request *http.
 	WriteData(response, http.StatusOK, mediaType, content)
 }
 
+type ImageIndex struct {
+	ispec.Index
+}
+
+func getReferrers(ctx context.Context, routeHandler *RouteHandler,
```

Comment:
this is not correct
1. we don't need to sync the subject, regctl will HEAD the subject before getting here, so at this point we can assume we already have the image synced on demand.

2. we need to update sync on demand to copy OCI referrers, like we do for oras. (the case where we have the image synced already but at a time when it didn't have referrers yet on upstream)

4. and also update sync to check each image it syncs if it has OCI referrers.

Reply:
@peusebiu pls fix this in a separate PR.